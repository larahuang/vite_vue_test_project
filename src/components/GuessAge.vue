<template>
  <h1 data-test="title" >{{ title }}</h1>
  <div class="card">
   <div style="margin-top: 35px;">
    <span>姓名: {{firstname}}</span> <br>
    <span>年齡: {{age}}</span> 

    </div>
    <div class="search_content">
     <label> Enter Firstname </label>
     <input type="text" 
     class="from_control"
      :value="search"
       @input="emitSearchValue"
     placeholder="搜尋姓名" 
     /> 
   <button 
   data-test="getAge"
   @click="getAge">獲取年齡</button>
   </div>
  </div>
</template>

<script setup lang="ts">
import { ref,computed } from 'vue'

const props = defineProps({ 
  title: { type: String },
   search: { type: String },
})
    const emit = defineEmits(['update:search'])
    const emitSearchValue = (evt:any) =>{ 
        emit('update:search', evt.target.value)
    }
   // const search = ref<string>('');
    const age = ref<string>('');
    const firstname = ref<string>('');
    const getAge = () => { 
        fetch('https://api.agify.io/?name='+ props.search)
            .then(response => response.json())
          .then(data => {
              console.log(data)
                age.value = data.age
                firstname.value = data.name
                props.search=""
            })
    }
</script>