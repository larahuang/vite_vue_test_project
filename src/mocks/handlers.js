//模擬 HTTP 請求
import { http, HttpResponse } from 'msw'
const api = "https://api.agify.io/";
export const restHandlers = [
   http.get(api, (req, res, ctx) =>
   {
      const query = {
           age: 55,
            name: "tope"
      }
      console.log('ctx.json',ctx.json([ query]))
      return res(ctx.status(200), ctx.json([
        query
      ]))
   }),
]
