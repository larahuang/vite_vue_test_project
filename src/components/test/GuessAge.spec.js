import { describe, it, expect, afterEach, beforeAll, afterAll, beforeEach } from 'vitest'
 import { setupServer } from 'msw/node'
import { mount } from "@vue/test-utils"
import GuessAge from "../GuessAge.vue"
//引入Api模擬
import { restHandlers }  from "../../mocks/handlers";

const server = setupServer(...restHandlers)
// 在所有測試之前啟動伺服器
beforeAll(() => server.listen({ onUnhandledRequest: 'error' }))
//  所有測試後關閉伺服器
afterAll(() => server.close())
// 每次測試後重置處理程序“對於測試隔離很重要”
afterEach(() => server.resetHandlers())

describe('GuessAge', () =>
{
    let wrapper
    const title = 'Guess User Age App'
    if (typeof document !== 'undefined') {
        beforeEach(() =>
        {
            wrapper = mount(GuessAge, {
                props: {
                    title
                }
            })
        })
    }
    it("測試 GuessAge 元件 props title", async () =>
    { 
        if (typeof document !== 'undefined') {
            expect(wrapper.find('[data-test="title"]').text()).toBe(title);
             
        }
    });
    it("測試資料是否為函數", () =>
    {
        expect(typeof GuessAge.data).toBe("undefined");
    });
    it('快照 UI 測試', () =>
    {
        if (typeof document !== 'undefined') {
            const wrapper = mount(GuessAge, {});
            expect(wrapper.text()).toMatchSnapshot()
        }
    })
    it("找到按鈕", () =>
    {
        if (typeof document !== 'undefined') {
          
            expect(wrapper.find('[data-test="getAge"]').exists()).toBe(true);
        }
    });

    it("按鈕點擊", async () =>
    {
        if (typeof document !== 'undefined') {
            const ac = await wrapper.get('[data-test="getAge"]').trigger("click")
            expect(wrapper.vm.search).toEqual("")
        }
    })
})


