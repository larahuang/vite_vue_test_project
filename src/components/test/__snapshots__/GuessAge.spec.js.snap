// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GuessAge > snapshot UI testing 1`] = `"姓名:  年齡:  Enter Firstname 獲取年齡"`;

exports[`GuessAge > 快照 UI 測試 1`] = `"姓名:  年齡:  Enter Firstname 獲取年齡"`;

exports[`snapshot UI testing 1`] = `"姓名:  年齡:  Enter Firstname 獲取年齡"`;
